photon-api/src/main/scala/com/linkedin/photon/ml/model/RandomEffectModel.scala

```diff
@@ -58,18 +56,27 @@ class RandomEffectModel(
 
     val currType = this.modelType
 
+    // TODO: Should verify that each update retains the model type, however this is slow and materializes the RDD early
+//    val isMatch = updatedModelsRdd
+//      .mapPartitions { iter =>
+//        Array(iter.forall(_._2.modelType == currType)).iterator
+//      }
+//      .fold(true) { case (result, nextVal) =>
+//        result && nextVal
+//      }
+//    require(isMatch, s"One or more models in updated RDD do not match model type '$currType'")
+
+    // Since model type matches, override explicit check using anonymous class
     new RandomEffectModel(updatedModelsRdd, randomEffectType, featureShardId) {
-
-      // TODO: The model types don't necessarily match, but checking each time is slow so copy the type for now
       override lazy val modelType: TaskType = currType
     }
   }
 
   /**
-   * Compute the score for the dataset.
+   * Compute the score for the data set.
```

Comment:
This looks strange to my eye.

Reply:
I'll change it back, but the correct spelling should be 'data set' and 'hyper-parameter'! ML engineers are destroying the English language.